### ISSUE TITLE
Change "mnei" to "neio"

### ISSUE THREAD

**Issue by finex** [wbolster/emacs-evil-colemak-basics#18]
Hi, I'm trying the colemak keybindings but I find more confortable using "neio" for navigation keys because "n" is the home key for the right hand and this allows to virtually map each finger to a movement.

Instead "mnei" force to use index finger for two navigation keys or shift the whole hand away from the home position destroying the benefits of colemak layout.

What do you think about?

**Comment by wbolster** [comment]
my honest opinion is that i completely disagree with you 🙃 

colemak `hnei` (qwerty `hjkl`) is actually quite ergonomic in vim-inspired layouts. the key idea is that the two strongest fingers are used for up/down navigation, and that's what many people often use, myself included.

personally i don't use colemak `h/i` (qwerty `h/l`) to go left/right very often. there are many other commands that are often better suited to what i'm trying to achieve:

- `b` / `w` (and the capital letter versions) to jump word by word
- `0` for beginning-of-line, or insert-at-beginning-of-line via capital `I` in qwerty vim (`U` with `evil-colemak-basics`)
- `$` for end-of-line, and `A` for insert-at-end-of-line

for many other operations like adding/changing/removing parentheses, `evil-surround` is much faster and quite natural once you're used to it, e.g. `ds(` to delete-surrounding-parens, `ysuw]` to surround-inner-word-square-parens, `cs({' to change-surround-parens-to-curly-brackets, etc.

in fact, the _starting point_ for the `evil-colemak-basics` package is the idea that `hnei` are ergonomic cursor keys, and everything else flows from that; see readme for details.

thanks for your message, feel free to discuss more. this is a rather niche topic and i'm passionate about hehe 🙃

**Comment by finex** [comment]
I completely agree to the second part, where you talk about all the better methods to navigate the file content. 

But, even if I agree to use the strongest fingers (index and middle finger) to the most used operation on file navigation (up/down), when you've to switch between buffers, for example, the concept is less valuable. I'm also using a tiling WM, where I've mapped "neio" to move between windows and I'd like to use the same movement keys on both WM and Emacs. I'm currently trying to remap **just** the movement keys to "neio". Let's see what it will happens.

Probably, if we want to find a more ergonomic key binding we should experimenting mappings like xah-fly-keys. Anyway we are extremely a niche, and we like it :100: 

Anyway thank you for your answer, you've make me think!

**Comment by wbolster** [comment]
👍🏼

this package was intentionally designed to be for ‘former qwerty+vim users who now use colemak+emacs+evil’ which obviously includes myself. i am still amazed that other people use it or at least get inspired by it. 🙃

anyway the idea was to sensibly minimise the key shuffling impact. switching to colemak is hard enough already.

redesigning the whole layout is an interesting thought experiment, sure, but to be honest actually learning+using that is not gonna happen for me.

i used qwerty+vim for 15 years followed by 6+ years (still ongoing) of colemak+evil, and i play multiple music instrument families 🎹🎷... there is a limit to what my fingers and brain can do 😜

**Comment by finex** [comment]
I've a different background: I've used vim as many years like you but only for very basic tasks so I don't have many problems to memorize new shortcuts. Also I've only been using Evil/Emacs for a couple of years and I've only scratched the surface. Probably this could allow to experiment a bit more. I'm also lost on the rabbit hole of a programmable mechanical keyboard... 

Anyway, thanks again for your answer. I will play a bit with your package and I will try to make a simpler one just to adapt the movement keys as I previously wrote. If I'll be happy with the result, I will gladly share the code.

P.S: I've seen the title change, good idea :+1:

**Comment by wbolster** [comment]
good for you, experimenting is fun 🙃

my muscle memory turned out to be almost hardwired in my brain, otherwise i probably would've tried to go for something simpler that is more colemak centric from the start, instead of the clear qwerty legacy that is very present in vim.

fyi: this package has multiple users, and 
 someone else took inspiration from this package before (see also its bug tracker, we communicated) before, it may inspire you to fall deeper into the 🐰🕳️ https://github.com/bmallred/evil-colemak-minimal

**Comment by XavRan** [comment]
I really love the work put into this package. The mapping makes a lot of sense and a lot of tools like snipe still works. But I'm also looking to change the motion keys to "neio". Mainly because I'm an ortho vertical staggered keyboard user. 
In this layout "neio" makes a lot more sense ergonomically. @finex where you able to achieve this configuration? :)

**Comment by finex** [comment]
> I really love the work put into this package. The mapping makes a lot of sense and a lot of tools like snipe still works. But I'm also looking to change the motion keys to "neio". Mainly because I'm an ortho vertical staggered keyboard user. In this layout "neio" makes a lot more sense ergonomically. @finex where you able to achieve this configuration? :)

Unfortunately not yet because I can't find a good balance between the various shortcuts.

**Comment by azzamsa** [comment]
Hi, @wbolster :wave: 

I've been using this package for over 3 years, and it has been a lifesaver.
Thank you so much for creating it.

At the same time, I've been using the [Miryoku](https://github.com/manna-harbour/miryoku) layout with `MIRYOKU_NAV=VI` to accommodate this package (`mnei`).
However, now I want to experiment and experience the pure/vanilla Miryoku layout (`neio`).
I've started using it in my Neovim.

I know you disagree with this change, and I don't intend to push my changes upstream (here).
But I'm curious: if you were to make this change, what additional remaps or changes would you need to make to accommodate `neio`?
So far, I've found that it clashes with the [`o` command, which appends (opens) a new line below the current line](https://vim.rtorr.com/).
What key would you suggest as a substitute? The only one I've thought of is `m/M`.

```diff
  | Colemak            | Qwerty             | Action                   | States |
  |--------------------|--------------------|--------------------------|--------|
  | `n`, `e`, `i`, `o` | `h`, `j`, `k`, `l` | navigate                 | `mnvo` |
  | `u`, `U`           | `i`, `I`           | insert                   | `_nv_` |
+ | `m`, `M`           | `o`, `O`           | append                   | `_nv_` |
```
